bigtable/bigtable.sh

```diff
@@ -0,0 +1,109 @@
+#!/bin/bash
+#    Copyright 2018 Google, Inc.
+#
+#    Licensed under the Apache License, Version 2.0 (the "License");
+#    you may not use this file except in compliance with the License.
+#    You may obtain a copy of the License at
+#
+#        http://www.apache.org/licenses/LICENSE-2.0
+#
+#    Unless required by applicable law or agreed to in writing, software
+#    distributed under the License is distributed on an "AS IS" BASIS,
+#    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#    See the License for the specific language governing permissions and
+#    limitations under the License.
+#
+# This actions installs cloud-bigtable-client (https://github.com/GoogleCloudPlatform/cloud-bigtable-client/)
+# on dataproc cluster and configure it to use cloud BigTable (https://cloud.google.com/bigtable/).
+
+set -euxo pipefail
+
+readonly HBASE_HOME='/usr/lib/hbase'
+readonly BIGTABLE_HBASE_CLIENT='bigtable-hbase-1.x-hadoop-1.3.0.jar'
+readonly BIGTABLE_HBASE_DL_LINK="http://central.maven.org/maven2/com/google/cloud/bigtable/bigtable-hbase-1.x-hadoop/1.3.0/${BIGTABLE_HBASE_CLIENT}"
+readonly SPARK_HBASE_CLIENT='shc-core-1.1.1-2.1-s_2.11.jar'
+readonly SPARK_HBASE_CLIENT_DL_LINK="http://repo.hortonworks.com/content/groups/public/com/hortonworks/shc-core/1.1.1-2.1-s_2.11/${SPARK_HBASE_CLIENT}"
+
+function update_apt_get() {
+  for ((i = 0; i < 10; i++)); do
+    if apt-get update; then
+      return 0
+    fi
+    sleep 5
+  done
+  return 1
+}
+
+function err() {
+  echo "[$(date +'%Y-%m-%dT%H:%M:%S%z')]: $@" >&2
+  return 1
+}
+
+function install_big_table_client() {
+  wget -q "${BIGTABLE_HBASE_DL_LINK}" -O "${HBASE_HOME}/lib/${BIGTABLE_HBASE_CLIENT}" \
+    || err 'Unable to install BigTable client libs.'
+  ln -s "${HBASE_HOME}/lib/${BIGTABLE_HBASE_CLIENT}" \
+    "/usr/lib/hadoop-mapreduce/lib/${BIGTABLE_HBASE_CLIENT}"
+}
+
+function install_shc() {
+  wget -q "${SPARK_HBASE_CLIENT_DL_LINK}" -O "/usr/lib/spark/external/${SPARK_HBASE_CLIENT}" \
+    || err 'Unable to install shc.'
+  ln -s "/usr/lib/spark/external/${SPARK_HBASE_CLIENT}" \
+    "/usr/lib/spark/external/shc-core.jar"
+}
+
+function configure_big_table_client() {
+
+#Update classpaths
+  cat << 'EOF' >> /etc/hadoop/conf/mapred-env.sh
+HADOOP_CLASSPATH="${HADOOP_CLASSPATH}:/usr/lib/hbase/*"
+HADOOP_CLASSPATH="${HADOOP_CLASSPATH}:/usr/lib/hbase/lib/*"
+HADOOP_CLASSPATH="${HADOOP_CLASSPATH}:/etc/hbase/conf"
+EOF
+  cat << 'EOF' >> /etc/spark/conf/spark-env.sh
+SPARK_DIST_CLASSPATH="${SPARK_DIST_CLASSPATH}:/usr/lib/hbase/*"
+SPARK_DIST_CLASSPATH="${SPARK_DIST_CLASSPATH}:/usr/lib/hbase/lib/*"
+SPARK_DIST_CLASSPATH="${SPARK_DIST_CLASSPATH}:/etc/hbase/conf"
+SPARK_DIST_CLASSPATH="${SPARK_DIST_CLASSPATH}:/usr/lib/spark/external/*"
```

Comment:
You should consider scoping this just to `shc*` or whatever the jar is called. Otherwise, you'll end up adding a bunch of unnecessary libraries to the classpath.

Reply:
You are right I will limit that to just shc-core.jar.